internal/brclient/brclient.go

```diff
@@ -20,35 +54,171 @@ const (
 	SanityValidation ValidationType = "sanity" // validation_sanity
 	// FullValidation does a complete validation of the etcd DB.
 	FullValidation ValidationType = "full" // validation_full
+	// httpClientRequestTimeout is the timeout for all requests made by the http client
+	httpClientRequestTimeout = 10 * time.Second
 )
 
 // BackupRestoreClient is a client to connect to the backup-restore HTTPs server.
 type BackupRestoreClient interface {
 	// GetInitializationStatus gets the latest state of initialization from the backup-restore.
-	GetInitializationStatus() InitStatus
+	GetInitializationStatus(ctx context.Context) (InitStatus, error)
 	// TriggerInitialization triggers the initialization on the backup-restore passing in the ValidationType.
-	TriggerInitialization(validationType ValidationType) error
+	TriggerInitialization(ctx context.Context, validationType ValidationType) error
 	// GetEtcdConfig gets the etcd configuration from the backup-restore, stores it into a file and returns the path to the file.
-	GetEtcdConfig() string
+	GetEtcdConfig(ctx context.Context) (string, error)
 }
 
+// brClient implements BackupRestoreClient interface.
 type brClient struct {
+	client             *http.Client
+	sidecarBaseAddress string
+	etcdConfigFilePath string
+}
+
+// NewDefaultClient creates a BackupRestoreClient using the SidecarConfig and etcd configuration at etcdConfigPath.
+// It delegates the responsibility to NewClient by passing in a default implementation of HttpClientCreator.
+func NewDefaultClient(sidecarConfig types.SidecarConfig, etcdConfigPath string) (BackupRestoreClient, error) {
+	client, err := createSidecarClient(sidecarConfig)
+	if err != nil {
+		return nil, err
+	}
+	return NewClient(client, sidecarConfig.GetBaseAddress(), etcdConfigPath)
+}
+
+// NewClient creates and returns a new BackupRestoreClient object
+func NewClient(httpClient *http.Client, sidecarBaseAddress string, etcdConfigFilePath string) (BackupRestoreClient, error) {
+	return &brClient{
+		client:             httpClient,
+		sidecarBaseAddress: sidecarBaseAddress,
+		etcdConfigFilePath: etcdConfigFilePath}, nil
+}
+
+func (c *brClient) GetInitializationStatus(ctx context.Context) (InitStatus, error) {
+	// create a child cancellable ctx for the http GET call.
+	httpCtx, cancelFn := context.WithCancel(ctx)
+	defer cancelFn()
+
+	// create request with ctx
+	req, err := http.NewRequestWithContext(httpCtx, http.MethodGet, c.sidecarBaseAddress+"/initialization/status", nil)
+	if err != nil {
+		return Unknown, fmt.Errorf("(GetInitializationStatus):  error creating http request: %v", err)
```

Comment:
Already logs do not contain function name but it has more context about error when it was happened [here](https://github.com/gardener/etcd-wrapper/blob/initial_development/internal/bootstrap/bootstrap.go#L74).

``` golang
if initStatus, err = i.brClient.GetInitializationStatus(ctx); err != nil {
    i.logger.Error("error while fetching initialization status", zap. Error(err))
}
```

Reply:
We have seen in etcd-backup-restore that logs are badly written and it makes it very difficult to find the origin of the log. Therefore you then establish a convention so that even if the logs are bad and are repeated across different functions you always prefix it with a function name so that you get a automatic namespace/context